packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -187,23 +187,31 @@ export const CouponForm = ({
         </h4>
       )}
       {hasCoupon ? (
-        <div className="flex" data-testid="coupon-hascoupon">
-          <div className="mr-4">{promotionCode}</div>
+        <div
+          className="flex justify-between items-center"
+          data-testid="coupon-hascoupon"
+        >
+          <div>{promotionCode}</div>
           {readOnly ? null : (
-            <button
-              className={`button ${readOnly ? 'hidden' : ''}`}
-              onClick={removeCoupon}
-              disabled={subscriptionInProgress}
-              data-testid="coupon-remove-button"
-            >
-              <Localized id="coupon-remove">
-                <span>Remove</span>
-              </Localized>
-            </button>
+            <div className="w-28 ml-4">
+              <button
+                className={`button ${readOnly ? 'hidden' : ''}`}
```

Comment:
Not sure what/if `readOnly` is checking here. I think we already check/abort a `readOnly` check above on L195 where if we _are_ `readOnly` we don't render this block, no?

Reply:
Nice catch! Thank you! Fixed.